include/seastar/rpc/rpc_impl.hh

```diff
@@ -209,6 +209,23 @@ struct serialize_helper<true> {
 template <typename Serializer, typename Output, typename... T>
 inline void do_marshall(Serializer& serializer, Output& out, const T&... args);
 
+// since C++23, std::apply() only accepts tuple-like types, while rpc::tuple is not
+// a tuple-like type. so we have to swing our own apply() which accepts rpc::tuple.
+template <typename Fn, typename Tuple, size_t ...Index>
+constexpr decltype(auto) apply_impl(Fn&& fn, Tuple&& t, std::index_sequence<Index...>) {
+    return std::invoke(
+        std::forward<Fn>(fn),
+        std::get<Index>(std::forward<Tuple>(t))...);
+}
+
+template <typename Fn, typename Tuple>
+constexpr decltype(auto) apply(Fn&& fn, Tuple&& t) {
+     return apply_impl(
+         std::forward<Fn>(fn),
+         std::forward<Tuple>(t),
+         std::make_index_sequence<std::tuple_size_v<std::remove_reference_t<Tuple>>>{});
```

Comment:
Couldn't we static_cast to std::tuple and forward to std::apply?

Reply:
ahh, we can! and probably we should . changed accordingly.